test/method_test.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017-2020 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2017 BaishanCloud. All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/ssl.h>
+
+#include "testutil.h"
+#include "../ssl/ssl_local.h"
+#include "ssltestlib.h"
+
+static char *cert = NULL;
+static char *pvk = NULL;
+
+struct subtest {
+	int min_version;
+	int max_version;
+	const SSL_METHOD* server_method;
+	const SSL_METHOD* client_method;
+};
+
+#define SUBTEST_TLS1_2_VERSION	0
+#define SUBTEST_TLS1_3_VERSION	1
+#define SUBTEST_QUANTITY		2
+
+static struct subtest subtests[] = {
+	{ TLS1_VERSION, TLS1_2_VERSION, NULL, NULL }, 
+	{ TLS1_VERSION, TLS1_3_VERSION, NULL, NULL }};
+
+static int test_method(int test)
+{
+    SSL_CTX *cctx = NULL, *sctx = NULL;
+    SSL *clientssl = NULL, *serverssl = NULL;
+    int testresult = 0;
+
+    if (!TEST_true(create_ssl_ctx_pair(NULL, TLS_server_method(), TLS_client_method(), subtests[test].min_version, subtests[test].max_version, &sctx, &cctx, cert, pvk))
+     || !TEST_true(create_ssl_objects(sctx, cctx, &serverssl, &clientssl, NULL, NULL)))
```

Comment:
Could you please split these lines and similar too long lines below?

Reply:
Done!